src/compute/src/compute_state.rs

```diff
@@ -432,8 +432,11 @@ impl<'a, A: Allocate> ActiveComputeState<'a, A> {
     pub fn report_compute_frontiers(&mut self) {
         let mut new_uppers = Vec::new();
 
+        // Maintain a single allocation for `new_frontier` to avoid allocating on every iteration.
+        let mut new_frontier = Antichain::new();
```

Comment:
No strong feelings, but I'd like to call out that this reverts a suggestion you made here: https://github.com/MaterializeInc/materialize/pull/20090#discussion_r1269855205 :D